desidlas/datasets/get_sightlines.py

```diff
@@ -0,0 +1,45 @@
+from desidlas.datasets.preprocess import estimate_s2n,normalize,rebin
+from desidlas.datasets.DesiMock import DesiMock
+from desidlas.dla_cnn.defs import best_v
+import numpy as np
+import os
+from os.path import join
+from pkg_resources import resource_filename
+
+
+datafile_path = os.path.join(resource_filename('desidlas', 'tests'), 'datafile')
```

Comment:
sorry for the confusion, these codes are already tested in test_preprocess.py. I just want to make these code in a py file because this is an important step in making the dataset(load fits file and store them in a npy file ). For the users who want use our code, they will just need to 'from get_sightlines.npy import get_sightlines'. Then the npy file can be generated when they input the file path of spectra

Reply:
sorry, I am still confused.  

At the very least, please put these lines into a new `def`

Reply:
excuse me, do you mean that I should make lines 7-14 in another py file but keep other lines still in get_sightline.py?

Reply:
What I mean is that those lines shouldn't be there at the start of a module like this.

They will be run every time the module is imported which shouldn't be required
and is prone to cause a crash (if those files were moved, renamed, etc.)

Reply:
Sorry I misunderstood you before. What I want to do with this get_sightlines.py file is for the people who want to use our codes, they will only need to run this file for the preprocessing. This may be more convenient to use because they don not need to check other py files for preprocessing. Maybe another way to achieve this goal is to remove this file and add a notebook. In the notebook we list every step for preprocessing just like in the test_preprocess.py . If the second way is better then I think I can just replace this get_sightlines.py with a notebook.

Reply:
I make these lines here because I try to include every step for preprocessing in this py file. Then this one file is enough for the people who just want to run this code. But now I realized maybe a notebook is more proper for this? If so I can remove this file to avoid confusion and crash.

Reply:
A Notebook sounds fine.  Thanks!

Reply:
OK, I will remove this file and upload a notebook

Reply:
I have edited get_sightlines.py and uploaded a notebook at desidlas/notebook to describe how to run the preprocessing codes